### PR TITLE
miscellaneous refactors

### PR DESCRIPTION
**Pull request by ebbit1q**
just some finds browsing the codebase

### CODE REVIEW

**Comment by ZeldaZach** [comment]
I do wonder the context behind some of these original bits

File: common/server_protocolhandler.cpp [hunk 1]
```diff
@@ -60,27 +60,27 @@ void Server_ProtocolHandler::prepareDestroy()
...
+        Server_Game *game = room->getGames().value(gameIterator.key());
+        if (!game) {
+            room->gamesLock.unlock();
             continue;
         }
-        g->gameMutex.lock();
-        Server_Player *p = g->getPlayers().value(gameIterator.value().second);
+        game->gameMutex.lock();
+        Server_Player *p = game->getPlayers().value(gameIterator.value().second);
         if (!p) {
```
Comment (tooomm) [inline common/server_protocolhandler.cpp:74]: Is there a reason you left `p` unchanged?

Comment (ebbit1q) [reply common/server_protocolhandler.cpp:74]: macros work very well for changes like this but don't have that kind of initiative

**Comment by ebbit1q** [comment]
> I do wonder the context behind some of these original bits

I did dig up the original reasoning for them but they've all been made obsolete by later changes.

**Review by ZeldaZach — APPROVED** [review]

**Pull request merged**